HermaFx.DataAnnotations/DistinctElementsAttribute.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace HermaFx.DataAnnotations
+{
+	[AttributeUsage(AttributeTargets.Property)]
+	public class DistinctElementsAttribute : ValidationAttribute
+	{
+		public string UniqueProperty { get; }
+		public Type ElementType { get; }
+		public DistinctElementsAttribute(string property)
+		{
+			ErrorMessage = "The property {0} must be unique in the collection.";
```

Comment:
Does the actual property collection name get shown somewhere in the thrown exception?

Reply:
You're right. I've added it.